### ISSUE TITLE
eth0 stops working after 1/2 day of uptime on bionic-minimal-rock64-0.6.31-209-arm64/mainline

### ISSUE THREAD

**Issue by seangineer** [ayufan-rock64/linux-build#158]
After about 6-12 hours of uptime on kernel 4.16.0-rockchip-ayufan-209-g027e722bb, the eth0 interface becomes unresponsive. `ip a` still lists a valid configuration, but there isn't any incoming or outgoing traffic.

A ping test of a hostname the following:

`root@rock64:~# ping google.com
ping: google.com: Temporary failure in name resolution
`

While pinging an ip address leads to this output:

```
root@rock64:~# ping 208.67.222.222
PING 208.67.222.222 (208.67.222.222) 56(84) bytes of data.
From 192.168.178.24 icmp_seq=1 Destination Host Unreachable
From 192.168.178.24 icmp_seq=2 Destination Host Unreachable
From 192.168.178.24 icmp_seq=3 Destination Host Unreachable
From 192.168.178.24 icmp_seq=4 Destination Host Unreachable
From 192.168.178.24 icmp_seq=5 Destination Host Unreachable
From 192.168.178.24 icmp_seq=6 Destination Host Unreachable
^C
--- 208.67.222.222 ping statistics ---
8 packets transmitted, 0 received, +6 errors, 100% packet loss, time 7157ms

```

The result is the same after disconnecting and reconnecting the Ethernet cable or restarting the interface using `systemctl restart NetworkManager`.

This is usually within a day of the full system becoming unresponsive as described in #157.

**Comment by seangineer** [comment]
On a potentially related note, I have noticed artifacts over SSH in the 0.6.XX builds on both mainline and the 4.4 kernels as shown below, though it is worse on mainline. 

I did not see this issue on 0.5.15 with the 4.4 kernel, but IIRC it was present on mainline.

Maybe this is related to the gmac stability issue? https://github.com/ayufan-rock64/linux-mainline-kernel/commit/73e42e18669934fa96cf2bb54291da54177076d7

![image](https://user-images.githubusercontent.com/7366784/38455771-33e3ced2-3a7d-11e8-9a1a-ae4877fc051f.png)
